### ISSUE TITLE
[Bug] Incorrect GUI Z Offset

### ISSUE THREAD

**Issue by Phoupraw** [GregTechCEu/GregTechCEu-1.19#265]
### GregTech CEu Version

fabric-1.20.1-1.0.10

### Recipe Viewer Installed

EMI

### Environment

Singleplayer

### Cross-Mod Interaction

Unsure

### Expected Behavior

In vanilla GUI, hovering stack should entirely cover stack in slot, and shouldn't render tooltip.
![W_VMH$ 4Z Y65K$V5YC`25B](https://github.com/GregTechCEu/GregTech-Modern/assets/50520903/b38f7efd-93af-4b43-b704-ada59ca5812d)

### Actual Behavior

In GT GUI, hovering stack is rendered at the middle z offset in the stack in slot, and its tooltip is rendered/
![{TSM)VDJT} YMIXQ1D0_%ZW](https://github.com/GregTechCEu/GregTech-Modern/assets/50520903/039c4510-8499-4b6a-9e0a-40e9d7a45fb3)
![RYE2BYV R~J WZU~}Y@A4GE](https://github.com/GregTechCEu/GregTech-Modern/assets/50520903/3730e957-47ed-4478-a7ca-0c09bc5e2a55)

### Steps to Reproduce

1. Get two stacks of items.
2. Hover one stack on the other stack.
3. Observe.

### Additional Information

This bug is found in development enviroment.

Loaded 74 mods:
  - ae2 15.0.4-beta
  - cardinal-components-base 5.2.1
  - cardinal-components-entity 5.2.1
  - cloth-config 11.0.99
  - com_github_llamalad7_mixinextras 0.2.0-beta.6
  - configuration 2.2.0
  - editenchanting 1.1.2
  - emi 1.0.18+1.20.1+fabric
  - enchantinginfuser 8.0.1
  - enchantist ${version}
  - fabric-api 0.83.1+1.20.1
  - fabric-api-base 0.4.29+b04edc7a77
  - fabric-api-lookup-api-v1 1.6.34+4d8536c977
  - fabric-biome-api-v1 13.0.10+b3afc78b77
  - fabric-block-api-v1 1.0.9+e022e5d177
  - fabric-blockrenderlayer-v1 1.1.39+b3afc78b77
  - fabric-client-tags-api-v1 1.0.20+b3afc78b77
  - fabric-command-api-v2 2.2.11+b3afc78b77
  - fabric-content-registries-v0 4.0.7+b3afc78b77
  - fabric-convention-tags-v1 1.5.3+b3afc78b77
  - fabric-crash-report-info-v1 0.2.18+aeb40ebe77
  - fabric-data-generation-api-v1 12.1.11+b3afc78b77
  - fabric-dimensions-v1 2.1.51+b3afc78b77
  - fabric-entity-events-v1 1.5.21+b3afc78b77
  - fabric-events-interaction-v0 0.6.0+b3afc78b77
  - fabric-game-rule-api-v1 1.0.38+b04edc7a77
  - fabric-gametest-api-v1 1.2.10+b3afc78b77
  - fabric-item-api-v1 2.1.26+b3afc78b77
  - fabric-item-group-api-v1 4.0.7+b3afc78b77
  - fabric-key-binding-api-v1 1.0.36+fb8d95da77
  - fabric-lifecycle-events-v1 2.2.20+b3afc78b77
  - fabric-loot-api-v2 1.1.37+b3afc78b77
  - fabric-message-api-v1 5.1.6+b3afc78b77
  - fabric-mining-level-api-v1 2.1.47+b3afc78b77
  - fabric-models-v0 0.3.35+b3afc78b77
  - fabric-networking-api-v1 1.3.8+b3afc78b77
  - fabric-object-builder-api-v1 11.0.6+b3afc78b77
  - fabric-particles-v1 1.0.28+b3afc78b77
  - fabric-recipe-api-v1 1.0.18+b3afc78b77
  - fabric-registry-sync-v0 2.2.6+b3afc78b77
  - fabric-renderer-api-v1 3.0.1+b3afc78b77
  - fabric-renderer-indigo 1.3.1+b3afc78b77
  - fabric-rendering-data-attachment-v1 0.3.33+b3afc78b77
  - fabric-rendering-fluids-v1 3.0.26+b3afc78b77
  - fabric-rendering-v1 3.0.6+b3afc78b77
  - fabric-resource-conditions-api-v1 2.3.5+ea08f9d877
  - fabric-resource-loader-v0 0.11.7+f7923f6d77
  - fabric-screen-api-v1 2.0.6+b3afc78b77
  - fabric-screen-handler-api-v1 1.3.27+b3afc78b77
  - fabric-sound-api-v1 1.0.12+b3afc78b77
  - fabric-transfer-api-v1 3.2.2+b3afc78b77
  - fabric-transitive-access-wideners-v1 4.2.0+b3afc78b77
  - fabricloader 0.14.21
  - forgeconfigapiport 8.0.0
  - gtceu 1.0.10
  - iceberg 1.1.13
  - jade 11.2.2
  - java 17
  - ldlib 1.0.16.b
  - minecraft 1.20.1
  - modmenu 7.1.0
  - porting_lib_core 2.1.1021+1.20
  - porting_lib_data 2.1.1021+1.20
  - porting_lib_gametest 2.1.1021+1.20
  - porting_lib_model_generators 2.1.1021+1.20
  - porting_lib_model_materials 2.1.1021+1.20
  - porting_lib_tags 3.0
  - porting_lib_utility 2.1.1021+1.20
  - puzzleslib 8.0.12
  - registrate-fabric 1.3.59-MC1.20
  - team_reborn_energy 2.3.0
  - travelersbackpack 1.20.1-9.1.2
  - universalenchants 8.0.0
    - extensibleenums 7.0.1

**Comment by Phoupraw** [comment]
I fixed it by mixin:
```java
@Mixin(AbstractFurnaceBlockEntity.class)
abstract class MAbstractFurnaceBlockEntity {

    @Unique
    private static final ThreadLocal<ItemStack> LOCAL_craftRecipe_itemStack2 = new ThreadLocal<>();
    
    @Redirect(method = "craftRecipe", at = @At(value = "INVOKE", target = "Lnet/minecraft/recipe/Recipe;getOutput(Lnet/minecraft/registry/DynamicRegistryManager;)Lnet/minecraft/item/ItemStack;"))
    private static ItemStack correctIncrement(Recipe<?> instance, DynamicRegistryManager dynamicRegistryManager) {
        var r = instance.getOutput(dynamicRegistryManager);
        LOCAL_craftRecipe_itemStack2.set(r);
        return r;
    }

    @Redirect(method = "craftRecipe", at = @At(value = "INVOKE", target = "Lnet/minecraft/item/ItemStack;increment(I)V"))
    private static void correctIncrement(ItemStack instance, int amount) {
        ItemStack output = LOCAL_craftRecipe_itemStack2.get();
        if (output == null) {
            instance.increment(amount);
        } else {
            instance.increment(output.getCount());
        }
    }

}
```

**Comment by Yefancy** [comment]
its a bug of Ldlib rendering, what is that mixin about?

**Comment by Phoupraw** [comment]
> its a bug of Ldlib rendering, what is that mixin about?

Sorry, I mistook issue. That mixin is for another issue.

**Comment by Yefancy** [comment]
fixed in the 1.0.11
